src/robot/libraries/String.py

```diff
@@ -261,6 +261,41 @@ def get_lines_matching_regexp(self, string, pattern):
         regexp = re.compile('^%s$' % pattern)
         return self._get_matching_lines(string, regexp.match)
 
+    def get_regexp_matches(self, string, pattern, *groups):
+        """Returns multiple match results of the given ``string`` that match the ``pattern``.
+        This keyword returns all matches(don't return on first match) and support group.
+        See the `Should Match Regexp` check how to use regex in Robot Framework.
+        If regexp doesn't match anything, return empty array. (Always Pass)
+        This will help you find in high volatility string.
+        Examples:
+        | ${no_match} | Get Regexp Matches | abcdefg123abcdefg123 | hijk |
+        | ${no_group} | Get Regexp Matches | abcdefg123abcdefg123 | abcdefg |
+        | ${group_regexp} | Get Regexp Matches | abcdefg123abcdefg123 | ab(?P<name>cd)e(fg) |
+        | ${with_group_name} | Get Regexp Matches | abcdefg123abcdefg123 | ab(?P<name>cd)e(fg) | name |
+        | ${with_group_names} | Get Regexp Matches | abcdefg123abcdefg123 | ab(?P<name>cd)e(?P<second_name>fg) | name | second_name |
+        | ${with_group_index} | Get Regexp Matches | abcdefg123abcdefg123 | ab(?P<name>cd)e(fg) | 2 |
+        | ${with_group_indexes} | Get Regexp Matches | abcdefg123abcdefg123 | ab(?P<name>cd)e(fg) | 1 | 2 |
+        | ${with_group_name_and_index} | Get Regexp Matches | abcdefg123abcdefg123 | ab(?P<name>cd)e(fg) | 2 | name |
+        =>
```

Comment:
Strings and patterns used in the examples are a bit long and complicated. Could you try simplifying them a little? I can look at that myself after the merge too.